### PR TITLE
Update supported ops md for NNAPI/CoreML EP

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
**Description**: Describe your changes.

As title.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

Update documentation based on newly added operator support.

### CODE REVIEW

File: tools/ci_build/github/android/nnapi_supported_ops.md [hunk 1]
```diff
@@ -10,12 +10,14 @@ Keep in sync with doco generated from /docs/execution-providers/NNAPI-ExecutionP
...
 |ai.onnx:Concat||
 |ai.onnx:Conv|Only 2D Conv is supported.<br/>Weights and bias should be constant.|
+|ai.onnx:DepthToSpace|Only DCR mode DepthToSpace is supported.|
 |ai.onnx:DequantizeLinear|All quantization scales and zero points should be constant.|
 |ai.onnx:Div||
 |ai.onnx:Elu||
 |ai.onnx:Exp||
 |ai.onnx:Flatten||
 |ai.onnx:Floor||
+|ai.onnx:Gather|Input indices should be an initializer if not int32 type.|
```
Comment (edgchen1) [inline tools/ci_build/github/android/nnapi_supported_ops.md:20]:
```suggestion
|ai.onnx:Gather|Input indices should be constant if not int32 type.|
```

File: tools/ci_build/github/android/nnapi_supported_ops.md [hunk 2]
```diff
@@ -43,6 +47,7 @@ Keep in sync with doco generated from /docs/execution-providers/NNAPI-ExecutionP
 |ai.onnx:Sub||
 |ai.onnx:Tanh||
 |ai.onnx:Transpose||
+|ai.onnx:Unsqueeze||
```
Comment (edgchen1) [inline tools/ci_build/github/android/nnapi_supported_ops.md:50]: axes must be constant

File: tools/ci_build/github/android/nnapi_supported_ops.md [hunk 3]
```diff
@@ -27,7 +29,9 @@ Keep in sync with doco generated from /docs/execution-providers/NNAPI-ExecutionP
 |ai.onnx:Min||
 |ai.onnx:Mul||
 |ai.onnx:Neg||
+|ai.onnx:Pad|Only constant mode Pad is supported.|
```
Comment (edgchen1) [inline tools/ci_build/github/android/nnapi_supported_ops.md:32]: also requires non-negative pad values and constant pad and input_value

File: tools/ci_build/github/apple/coreml_supported_ops.md [hunk 4]
```diff
@@ -11,15 +11,17 @@ Keep in sync with doco generated from /docs/execution-providers/CoreML-Execution
 |ai.onnx:Clip||
 |ai.onnx:Concat||
 |ai.onnx:Conv|Only 1D/2D Conv is supported.<br/>Weights and bias should be constant.|
+|ai.onnx:DepthToSpace|Only DCR mode DepthToSpace is supported.|
 |ai.onnx:Gemm|Input B should be constant.|
 |ai.onnx:GlobalAveragePool|Only 2D Pool is supported.|
 |ai.onnx:GlobalMaxPool|Only 2D Pool is supported.|
 |ai.onnx:MatMul|Input B should be constant.|
 |ai.onnx:MaxPool|Only 2D Pool is supported.|
+|ai.onnx:PRelu|Input slope should be a constant initializer.<br/>Input slope should either have shape [C, 1, 1] or have 1 element.|
```
Comment (edgchen1) [inline tools/ci_build/github/apple/coreml_supported_ops.md:20]:
```suggestion
|ai.onnx:PRelu|Input slope should be constant.<br/>Input slope should either have shape [C, 1, 1] or have 1 element.|
```

**Comment by edgchen1** [comment]
thanks for updating these!

**Comment by YUNQIUGUO** [comment]
> thanks for updating these!

thanks for the review. missed a couple points. updated now!

File: tools/ci_build/github/android/nnapi_supported_ops.md [hunk 5]
```diff
@@ -39,10 +43,11 @@ Keep in sync with doco generated from /docs/execution-providers/NNAPI-ExecutionP
 |ai.onnx:Slice||
 |ai.onnx:Softmax||
 |ai.onnx:Sqrt||
-|ai.onnx:Squeeze||
+|ai.onnx:Squeeze|Input axes must be constant.|
```
Comment (edgchen1) [inline tools/ci_build/github/android/nnapi_supported_ops.md:46]: nit: "should" for consistency with other notes. sorry for the other wording in the original comment.
```suggestion
|ai.onnx:Squeeze|Input axes should be constant.|
```

**Review by edgchen1 — DISMISSED** [review]

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**